app/controllers/amp_articles_controller.rb

```diff
@@ -0,0 +1,47 @@
+class AmpArticlesController < ActionController::Base
+  def show
+    if (@article = Article.find_by_permalink(params[:from])).present?
+      if @article.supports_amp || params[:no_redirect]
```

Comment:
Tiny, tiny thing, but for boolean columns it's nice to use the reader with the question mark on the end - `@article.supports_amp?`.